File: sdks/csharp/src/HookSniff/Resources/Analytics.cs
using HookSniff.Model;

namespace HookSniff.Resources;

/// <summary>
/// Analytics resource — delivery trends, success rate, latency.
/// </summary>
public class Analytics
{
    private readonly RequestContext _ctx;

    internal Analytics(RequestContext ctx) => _ctx = ctx;

    /// <summary>Get delivery trends.</summary>
    public async Task<DeliveryTrendResponse> TrendsAsync(string? since = null, string? until = null)
    {
        var req = new Request("GET", "/v1/analytics/deliveries");
        var qp = new Dictionary<string, object?>();
        if (since != null) qp["since"] = since;
        if (until != null) qp["until"] = until;
        req.SetQueryParams(qp);
        return (await req.SendAsync<DeliveryTrendResponse>(_ctx))!;
    }

    /// <summary>Get success rate.</summary>
    public async Task<SuccessRateResponse> SuccessRateAsync()
    {
        var req = new Request("GET", "/v1/analytics/success-rate");
        return (await req.SendAsync<SuccessRateResponse>(_ctx))!;
    }

    /// <summary>Get latency data.</summary>
    public async Task<LatencyResponse> LatencyAsync()
    {
        var req = new Request("GET", "/v1/analytics/latency");
        return (await req.SendAsync<LatencyResponse>(_ctx))!;
    }
}

File: sdks/python/hooksniff/resources/analytics.py
"""
HookSniff SDK — Analytics Resource
"""

import typing as t

from ..models import (
    StatsResponse,
    TrendResponse,
    SuccessRateResponse,
    LatencyResponse,
    DeliveryTrendResponse,
    TrendPoint,
)
from .common import ApiBase


class AnalyticsAsync(ApiBase):
    async def get_stats(self, period: str = "30d") -> StatsResponse:
        """Get overall stats for the given period."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/analytics/stats",
            query_params={"period": period},
        )
        return StatsResponse(**response.json())

    async def get_trends(self, metric: str = "deliveries", period: str = "30d") -> TrendResponse:
        """Get trend data for a metric."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/analytics/trends",
            query_params={"metric": metric, "period": period},
        )
        return TrendResponse(**response.json())

    async def get_success_rate(self, period: str = "30d") -> SuccessRateResponse:
        """Get success rate overall and by endpoint."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/analytics/success-rate",
            query_params={"period": period},
        )
        return SuccessRateResponse(**response.json())

    async def get_latency(self, period: str = "30d") -> LatencyResponse:
        """Get latency trends (p50, p95, p99)."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/analytics/latency",
            query_params={"period": period},
        )
        return LatencyResponse(**response.json())

    async def get_delivery_trends(self, period: str = "30d") -> DeliveryTrendResponse:
        """Get delivery volume trends."""
        response = await self._request_asyncio(
            method="get",
            path="/api/v1/analytics/delivery-trends",
            query_params={"period": period},
        )
        return DeliveryTrendResponse(**response.json())


class Analytics(ApiBase):
    def get_stats(self, period: str = "30d") -> StatsResponse:
        """Get overall stats."""
        response = self._request_sync(
            method="get",
            path="/api/v1/analytics/stats",
            query_params={"period": period},
        )
        return StatsResponse(**response.json())

    def get_trends(self, metric: str = "deliveries", period: str = "30d") -> TrendResponse:
        """Get trend data."""
        response = self._request_sync(
            method="get",
            path="/api/v1/analytics/trends",
            query_params={"metric": metric, "period": period},
        )
        return TrendResponse(**response.json())

    def get_success_rate(self, period: str = "30d") -> SuccessRateResponse:
        """Get success rate."""
        response = self._request_sync(
            method="get",
            path="/api/v1/analytics/success-rate",
            query_params={"period": period},
        )
        return SuccessRateResponse(**response.json())

    def get_latency(self, period: str = "30d") -> LatencyResponse:
        """Get latency trends."""
        response = self._request_sync(
            method="get",
            path="/api/v1/analytics/latency",
            query_params={"period": period},
        )
        return LatencyResponse(**response.json())

    def get_delivery_trends(self, period: str = "30d") -> DeliveryTrendResponse:
        """Get delivery volume trends."""
        response = self._request_sync(
            method="get",
            path="/api/v1/analytics/delivery-trends",
            query_params={"period": period},
        )
        return DeliveryTrendResponse(**response.json())
